src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java

```diff
@@ -128,4 +129,46 @@ public boolean supportsFolder() {
         return true;
     }
 
+
+    /**
+     * @deprecated Use <tt>GlobalConfigFiles.get().getConfigs()</tt> instead.
+     */
+    @Deprecated
+    public Collection<Config> getAllConfigs() {
```

Comment:
If you think this is a good solution for the openstack extention of the config-file-provider, then I think you should just access `GlobalConfigFiles.get()` directly